spaceship/views.py

```diff
@@ -68,17 +68,36 @@ def home():
 
 @app.route('/login', methods=['GET', 'POST'])
 def login():
+  oauth_user_info = session.pop('oauth_user_info', None)
   if current_user.is_authenticated:
     return redirect_for_logged_in()
 
+  session['oauth_next_url'] = url_for('login')
```

Comment:
should we only set this when we are going to render the login template?

Reply:
yes good idea, done